azure-kusto-ingest/azure/kusto/ingest/_resource_manager.py

```diff
@@ -131,3 +134,15 @@ def get_ingestions_status_tables(self) -> List[_ResourceUri]:
     def get_authorization_context(self):
         self._refresh_authorization_context()
         return self._authorization_context
+
+    def retrieve_service_type(self):
+        try:
+            command_result = self._kusto_client.execute("NetDefaultDB", _SHOW_VERSION)
+        except KustoServiceError:
+            raise KustoServiceError("Couldn't retrieve ServiceType because of a service exception executing {0}".format(_SHOW_VERSION), None)
```

Comment:
I thought rethrowing a cleaner error would simplify and thus best help the SDK user, but perhaps it's best not to get involved. I'm removing these so that whatever error is thrown will be raised without my simplified explanations.

Reply:
You misunderstand,  
The clear error is for the SDK user.
The encapsulated error is for us to be able to assist him when he has a problem.
The user does not care about the fact that you implicitly called .show version, but we care very much about why it had failed.
Different audience, different needs.

Reply:
No I understand that we can try to address different audiences, but the fact is that:
1) The SDK user will see both exceptions if we rethrow while including the original exception instead of swallowing it
2) The SDK user will have to follow the same next steps either way. They will have to either debug/look deeper into the code, or open a support ticket.
For both of these reasons, adding a simplified message is not necessary, not a big benefit, and complicates the review.